src/library_pthread.js

```diff
@@ -314,7 +314,12 @@ var LibraryPThread = {
             } else if (d.cmd === 'exitProcess') {
               // A pthread has requested to exit the whole application process (runtime).
               Module['noExitRuntime'] = false;
-              exit(d.returnCode);
+              try {
+                exit(d.returnCode);
```

Comment:
perhaps there is something more suitable we can call here?  pthread_exit?   I'm not sure calling the user-level exit API makes sense.

Reply:
I assume the intention is still to run `atexit` and `Module.onExit` handlers as applicable?

Reply:
I think this is basically a proxied `exit()` to the main thread? So it seems correct to me.

Reply:
BTW would this result in atexit handlers getting called more than once? If they are supposed to get called on the thread that calls exit(), then we wouldn't want this. But what about `Module.onExit` handlers? I guess they would get called on the thread that registered them in JS in all cases. So maybe we need to split that here.
Without PROXY_TO_PTHREAD, do atexit handlers + pthreads make any sense? There's no way to spawn a pthread and wait without returning to the main loop, so you'd need EXIT_RUNTIME=0 anyway, right?

Reply:
Yes, good point, in pthreads the main loop can't really exit and wait for a pthread - it can't run synchronously forever, and if it waits asynchronously then the runtime must not be exited.

In theory atexits matter if main never waits for a pthread, so it runs synchronously for a short time and the pthreads also finish in that time.

If it's simplest, not support atexits with pthreads seems safe to me for now.

Reply:
I investigated the current state of `atexit` vs pthreads in #9251 and it's not pretty. Probably we should fix it at some point, perhaps along with a cleanup of the pthreads JS code after we remove fastcomp. But this PR doesn't change that current behavior and it will make it easier to detect local test failures, so I'm going to land it for now.